plugins/Live/Visitor.php

```diff
@@ -315,27 +316,21 @@ public static function enrichVisitorArrayWithActions($visitorDetailsArray, $acti
         return $visitorDetailsArray;
     }
 
-    private static function sortByServerTime($a, $b)
+    private static function sortActionDetails($actions)
     {
-        $ta = strtotime($a['serverTimePretty']);
-        $tb = strtotime($b['serverTimePretty']);
-
-        if ($ta < $tb) {
-            return -1;
-        }
-
-        if ($ta == $tb) {
-            if ($a['idlink_va'] == $b['idlink_va']) {
-                return strcmp($a['type'], $b['type']);
-            }
-
-            if ($a['idlink_va'] > $b['idlink_va']) {
-               return 1;
+        usort($actions, function ($a, $b) {
+            $fields = array('serverTimePretty', 'idlink_va', 'title', 'url', 'type', 'pageIdAction', 'goalId');
```

Comment:
eg https://builds-artifacts.matomo.org/matomo-org/matomo/4.x-dev/37985/UIIntegrationTest_widgetize_visitor_log.png was still failing randomly cause there are 2 goal conversions with same time and idlinkva. Only the goal ID was different

```sql
INSERT INTO `log_conversion` (`idvisit`, `idsite`, `idvisitor`, `server_time`, `idaction_url`, `idlink_va`, `idgoal`, `buster`, `idorder`, `items`, `url`, `visitor_days_since_first`, `visitor_days_since_order`, `visitor_returning`, `visitor_count_visits`, `referer_keyword`, `referer_name`, `referer_type`, `config_device_brand`, `config_device_model`, `config_device_type`, `location_city`, `location_country`, `location_latitude`, `location_longitude`, `location_region`, `revenue`, `revenue_discount`, `revenue_shipping`, `revenue_subtotal`, `revenue_tax`, `custom_var_k1`, `custom_var_v1`, `custom_var_k2`, `custom_var_v2`, `custom_var_k3`, `custom_var_v3`, `custom_var_k4`, `custom_var_v4`, `custom_var_k5`, `custom_var_v5`, `custom_dimension_1`, `custom_dimension_2`, `custom_dimension_3`, `custom_dimension_4`, `custom_dimension_5`)
VALUES
	(57, 1, X'D0966C8CDB89610B', '2012-09-30 08:10:38', 1, 70, 1, 0, NULL, NULL, 'http://piwik.net/blog/category/meta/', 0, 0, 0, 1, NULL, NULL, 1, '', '', NULL, NULL, 'jp', 36.000000, 138.000000, NULL, 5, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL),
	(57, 1, X'D0966C8CDB89610B', '2012-09-30 08:10:38', 1, 70, 2, 0, NULL, NULL, 'http://piwik.net/blog/category/meta/', 0, 0, 0, 1, NULL, NULL, 1, '', '', NULL, NULL, 'jp', 36.000000, 138.000000, NULL, 5, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
```